src/data_server/utils/user_client.rs

```diff
@@ -0,0 +1,53 @@
+use aruna_rust_api::aruna::aruna::api::storage::services::v1::collection_service_client;
+use tonic::codegen::InterceptedService;
+use tonic::metadata::AsciiMetadataKey;
+use tonic::metadata::AsciiMetadataValue;
+use tonic::transport::{Channel, ClientTlsConfig};
+
+const API_TOKEN_ENTRY_KEY: &str = "Authorization";
+
+#[derive(Clone)]
+pub struct UserClient {
+    pub collection_service: collection_service_client::CollectionServiceClient<
+        InterceptedService<Channel, ClientInterceptor>,
+    >,
+}
+
+#[derive(Clone)]
+pub struct ClientInterceptor {
+    api_token: String,
+}
+
+impl UserClient {
+    pub async fn new(endpoint: String, api_token: String) -> Self {
+        let interceptor = ClientInterceptor { api_token };
+        let tls_config = ClientTlsConfig::new();
+        let endpoint = Channel::from_shared(endpoint)
+            .unwrap()
```

Comment:
please no unwraps here, additionally it is better to create a new connection each time a "user-client" request is send. Infrequent requests will otherwise get timeouts due to unused connections.